File: datasets/lua-keep-incoder_6b_temp_0.2-1/HumanEval_26_remove_duplicates.lua
-- From a list of integers, remove all elements that occur more than once.
-- Keep order of elements left the same as in the input.
-- >>> remove_duplicates([1, 2, 3, 2, 4])
-- [1, 3, 4]
local function remove_duplicates(numbers)
  local result = {}
  for _, value in ipairs(numbers) do
    if result[value] == nil then
      result[value] = value
    end
  end
  return result
end


lu = require('luaunit')

function test_humaneval()
local candidate = remove_duplicates
    lu.assertEquals(candidate({}), {})
    lu.assertEquals(candidate({1, 2, 3, 4}), {1, 2, 3, 4})
    lu.assertEquals(candidate({1, 2, 3, 2, 4, 3, 5}), {1, 4, 5})
end

os.exit(lu.LuaUnit.run())

File: datasets/R-keep-code_davinci_001_temp_0.2/HumanEval_26_remove_duplicates.R
# From a list of integers, remove all elements that occur more than once.
# Keep order of elements left the same as in the input.
# >>> remove_duplicates([1, 2, 3, 2, 4])
# [1, 3, 4]
remove_duplicates <- function(numbers) {
    unique_numbers <- unique(numbers)
    unique_numbers[duplicated(unique_numbers)] <- NULL
    return(unique_numbers)
}


test_humaneval <- function() {
candidate <- remove_duplicates
    if(!identical(candidate(c()), c())){quit('no', 1)}
    if(!identical(candidate(c(1, 2, 3, 4)), c(1, 2, 3, 4))){quit('no', 1)}
    if(!identical(candidate(c(1, 2, 3, 2, 4, 3, 5)), c(1, 4, 5))){quit('no', 1)}
}
test_humaneval()